torch/testing/_internal/optests/aot_autograd.py

```diff
@@ -0,0 +1,112 @@
+import torch
+import torch.utils._pytree as pytree
+from torch.testing._internal.common_methods_invocations import wrapper_set_seed
+from functorch.compile import compiled_function, min_cut_rematerialization_partition, nop
+import re
+
+
+class assert_raises_regex:
+    def __init__(self, exception_cls, regex):
+        self.exception_cls = exception_cls
+        self.regex = regex
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, traceback):
+        if exc_type == self.exception_cls:
+            msg = str(exc_val)
+            if not re.search(self.regex, msg):
+                raise AssertionError(
+                    f"Expected exception to match regex. regex: {self.regex}, exception: {msg}")
+            return True  # Squashes the exception
+        if exc_type is not None:
+            raise AssertionError(
+                f"Expected {self.exception_cls} to be raised, instead got exception {exc_type}")
+        raise AssertionError("Expected exception to be raised but none was")
+
+
+def aot_autograd_check(
+        func, args, kwargs, dynamic,
+        assert_raises_regex_fn=assert_raises_regex,
+        assert_equals_fn=torch.testing._comparison.assert_close):
+    flat_args, args_spec = pytree.tree_flatten((args, kwargs))
+    for arg in flat_args:
+        if isinstance(arg, int) and arg == -42:
+            raise TestFrameworkError("We've reserved -42 as an arg")
+
+    sentinel_val = -42
+    is_tensor_spec = [sentinel_val if isinstance(arg, torch.Tensor) else arg for arg in flat_args]
```

Comment:
still reading, but what prevents us from using the original flat_args, so we need the sentinal value?

nit: probably rename `is_tensor_spec` -> `flat_args_with_sentinel`?

Reply:
> still reading, but what prevents us from using the original flat_args, so we need the sentinal value?

This was in the original code that I moved here. I'm happy to make improvements to it. We don't need to make `is_tensor_spec`, instead we should just store a bool of True/False of if the thing is a Tensor or not and use the original flat_args